shapely/tests/test_flatcoords.py

```diff
@@ -0,0 +1,351 @@
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose
+
+import shapely
+from shapely import MultiLineString, MultiPoint, MultiPolygon
+from shapely.testing import assert_geometries_equal
+
+from .common import (
+    geometry_collection,
+    line_string,
+    line_string_z,
+    linear_ring,
+    multi_line_string,
+    multi_line_string_z,
+    multi_point,
+    multi_point_z,
+    multi_polygon,
+    multi_polygon_z,
+    point,
+    point_z,
+    polygon,
+    polygon_z,
+)
+
+all_types = (
+    point,
+    line_string,
+    polygon,
+    multi_point,
+    multi_line_string,
+    multi_polygon,
+)
+
+all_types_3d = (
+    point_z,
+    line_string_z,
+    polygon_z,
+    multi_point_z,
+    multi_line_string_z,
+    multi_polygon_z,
+)
+
+all_types_not_supported = (
+    linear_ring,
+    geometry_collection,
+)
+
+
+@pytest.mark.parametrize("geom", all_types)
+def test_roundtrip(geom):
+    actual = shapely.from_ragged_array(*shapely.to_ragged_array([geom, geom]))
+    assert_geometries_equal(actual, [geom, geom])
+
+
+@pytest.mark.parametrize("geom", all_types_3d)
+def test_roundtrip_3d(geom):
+    actual = shapely.from_ragged_array(
+        *shapely.to_ragged_array([geom, geom], include_z=True)
+    )
+    assert_geometries_equal(actual, [geom, geom])
+
+
+@pytest.mark.parametrize("geom", all_types)
+def test_include_z(geom):
+    _, coords, _ = shapely.to_ragged_array([geom, geom], include_z=True)
+    # For 2D geoms, z coords are filled in with NaN
+    assert np.isnan(coords[:, 2]).all()
+
+
+@pytest.mark.parametrize("geom", all_types_3d)
+def test_include_z_false(geom):
+    _, coords, _ = shapely.to_ragged_array([geom, geom], include_z=False)
+    # For 3D geoms, z coords are dropped
+    assert coords.shape[1] == 2
+
+
+@pytest.mark.parametrize("geom", all_types_not_supported)
+def test_raise_geometry_type(geom):
+    with pytest.raises(ValueError):
+        shapely.to_ragged_array([geom, geom])
+
+
+def test_points():
+    arr = shapely.from_wkt(
```

Comment:
Suggest using `shapely.points` constructor here instead:

```
shapely.points([[0, 0], [1, 1], [np.nan, np.nan], ...])
```

(makes it a little easier to compare to `expected` below)

In fact, you could probably move `expected` up here, then

```Python
points = shapely.points(expected)
```

Reply:
> Suggest using `shapely.points` constructor here instead:

Unfortunately that doesn't work reliably for empty geometries (you would have to use NaNs, but that depends on the GEOS version whether they get turned into empty geometries or not). 
(I would prefer to keep `expected` independent from the input)

I have been trying some different ways, but in the end landed on WKT seemingly being the most robust way to represent all kinds of geometries in a human readable way. 
I am also planning to create a standard test suite in geoarrow/geoarrow and use WKT there as well. And then it would be nice to keep the data consistent.